pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -716,13 +720,23 @@ func (dm *DockerManager) runContainer(
 
 // setInfraContainerNetworkConfig sets the network configuration for the infra-container. We only set network configuration for infra-container, all
 // the user containers will share the same network namespace with infra-container.
-func setInfraContainerNetworkConfig(pod *api.Pod, netMode string, opts *kubecontainer.RunContainerOptions, dockerOpts *dockertypes.ContainerCreateConfig) {
+func (dm *DockerManager) setInfraContainerNetworkConfig(pod *api.Pod, netMode string, opts *kubecontainer.RunContainerOptions, dockerOpts *dockertypes.ContainerCreateConfig) {
 	exposedPorts, portBindings := makePortsAndBindings(opts.PortMappings)
 	dockerOpts.Config.ExposedPorts = exposedPorts
 	dockerOpts.HostConfig.PortBindings = dockernat.PortMap(portBindings)
 
+	result, err := dm.checkDockerAPIVersion(dockerV112APIVersion)
+	if err != nil {
+		glog.Errorf("Failed to check docker api version: %v", err)
+	}
+
+	hostNetPlusDNSEnabled := result >= 0
```

Comment:
If check version failed, the result is also `0`. We may not want to use new feature at that time.